code/modules/atmospherics/machinery/components/unary_devices/cryo.dm

```diff
@@ -89,36 +89,41 @@
 		beaker = null
 
 /obj/machinery/atmospherics/components/unary/cryo_cell/update_icon()
+
 	cut_overlays()
 
 	if(panel_open)
 		add_overlay("pod-panel")
 
-	if(state_open)
+	if(state_open && (icon_state != "pod-open"))

```

Comment:
Do not test `icon_state`, add some control var instead.

Reply:
Why wouldnt i check icon state in a proc designated specifically to managing icon states in an environment where we do not track icon states in any specific way besides its factual value?

Reply:
It's a bad practice, icon_states shouldn't be used for logic (even if it's icon logic), only for output.

Reply:
Adding a control var for something which should be handled naturally should we not opted for universal cut and rebuild proc on every occasion is as bad of a cancer as switching machinery on or off based on icon_state

Reply:
Adding a var for logic makes code more maintainable dude, if you check for `icon_state` it'll be painful and error-prone for someone to change the default open `icon_state`

Reply:
Adding a var that traces a very specific transition state from on to off is just as bad
At best its an argument to update_icons() with 3 states: swtiching_to_open, switching_to_close, non_switching_update
But that would need some additional proc call chain changes since machines all use generic open and close procs and they call a generic update_icons there